source/configuration_manual/forwarding_parameters.rst

```diff
@@ -1,74 +1,85 @@
 .. _forwarding_parameters:
 
 =====================================================
-Forwarding Parameters in IMAP/POP3/LMTP/SMTP proxying
+Forwarding parameters in IMAP/POP3/LMTP/SMTP proxying
 =====================================================
 
-Dovecot supports proxying various pieces of information and even variables for various protocols when forwarding connection. It requires that the sender is listed under ``trusted_networks``. For IMAP, it uses the ``ID`` command, for other protocols, ``XCLIENT`` is used.
+Dovecot supports proxying various pieces of information and even
+variables for various protocols when forwarding connection. It requires
+that the sender is listed under ``trusted_networks``. For IMAP, it uses
+the ``ID`` command, for other protocols, ``XCLIENT`` is used.
 
-This feature is supported since v1.2, except for x-forward which was added in v2.2.29.
+This feature is supported since v1.2, except for x-forward which was
+added in v2.2.29.
 
 IMAP protocol
-=============
+-------------
 
-For IMAP protocol, this is done by extending the ID command. The maximum length of parameters to parse is 255 bytes. The parameters are forwarded as part of the ID command key-value list.
+For IMAP protocol, this is done by extending the ID command. The maximum
+length of parameters to parse is 255 bytes. The parameters are forwarded
+as part of the ID command key-value list.
```

Comment:
* Maximum key length is 30 bytes.
 * RFC says:
   * Value strings MUST NOT be longer than 1024 octets.
   * Implementations MUST NOT send more than 30 field-value pairs.
 * We have imaptest "id" test that verifies that Dovecot accepts ID command with all these limits.
 * Dovecot appears to have exactly 1024 byte limit to values, although I'm not exactly sure it takes that from. Trying to send 1025 bytes results in "BYE Input buffer full, aborting"
 * There doesn't seem to be any limit to number of key-value pairs. I sent hundreds of thousands and Dovecot happily read them. It didn't result in any increased memory usage, since each key-value pair was read separately and then discarded when not used.